enterprise/frontend/src/metabase-enterprise/whitelabel/components/IllustrationWidget/IllustrationWidget.tsx

```diff
@@ -0,0 +1,193 @@
+import type { ChangeEvent } from "react";
+import { useMemo, useState, useRef } from "react";
+import { createPortal } from "react-dom";
+import { t } from "ttag";
+
+import { color } from "metabase/lib/colors";
+import { Button, Flex, Icon, Paper, Select, Text } from "metabase/ui";
+import type {
+  EnterpriseSettingKey,
+  EnterpriseSettings,
+} from "metabase-enterprise/settings/types";
+
+import { LighthouseImage, PreviewImage } from "./IllustrationWidget.styled";
+
+type IllustrationValue = "default" | "no-illustration" | "custom";
+
+export interface StringSetting {
```

Comment:
I could use the name of the widget instead like this example https://github.com/metabase/metabase/blob/51f9caf975b7368b52649aefd8594b52e71ae0ac/enterprise/frontend/src/metabase-enterprise/whitelabel/components/LighthouseToggleWidget/types.ts#L1-L4

so we could use `IllustrationSetting` or something. The fact that I didn't want to use that because I don't really care what setting this is, in fact, I only care that the `setting.value` is a string so I named it that way.

For me if we were to name something like `IllustrationSetting` or `<insert actual setting name>Setting` I'd think it's the type of the whole setting instead. Something like
https://github.com/metabase/metabase/blob/51f9caf975b7368b52649aefd8594b52e71ae0ac/enterprise/frontend/src/metabase-enterprise/whitelabel/index.js#L207-L222

So... What do you think?

Reply:
The thing is that I read "StringSetting", but then the actual value is IllustrationValue, even though technically the underlaying type is a string (union)

Maybe to avoid having to come up with names, we could think about having a wrapper generic type, such as `SettingValue<T>` or `SettingValueWithDefault<T>` but that's something we'd have to chat about with team webapp, for this PR it's ok as it is now 👍

Reply:
I like the generic solution. Could be a good option going forward, but agree let's not doing anything in this PR right now.